sdk/python/kfp/compiler/_op_to_template.py

```diff
@@ -287,7 +288,10 @@ def _op_to_template(op: BaseOp):
         template.setdefault('metadata', {}).setdefault('annotations', {})['pipelines.kubeflow.org/task_display_name'] = processed_op.display_name
 
     if isinstance(op, dsl.ContainerOp) and op._metadata:
-        import json
         template.setdefault('metadata', {}).setdefault('annotations', {})['pipelines.kubeflow.org/component_spec'] = json.dumps(op._metadata.to_dict(), sort_keys=True)
 
+    if isinstance(op, dsl.ContainerOp) and op.execution_options:
+        if op.execution_options.caching_strategy.max_cache_staleness:
+            template.setdefault('metadata', {}).setdefault('annotations', {})['max_cache_staleness'] = str(op.execution_options.caching_strategy.max_cache_staleness)
```

Comment:
should we add a kfp prefix on annotation key: 'pipelines.kubeflow.org/max_cache_staleness'

Reply:
I'm on the fence on this - whether the caching is limited to KFP or has wider scope.

Reply:
Done.